httpc.c

```diff
@@ -1258,7 +1258,7 @@ static int httpc_state_machine(httpc_t *h, const char *url, int op) {
 	}
 	case SM_RCVB:
 		next = SM_DONE;
-		if (op == HTTPC_GET) {
+		if (h->rcv) {
```

Comment:
this condition is suboptimal, in case of response `204 No Content` there is no point reading body, because there is none.

Reply:
Looks like it can still return something: https://www.rfc-editor.org/rfc/rfc2616#section-10.2.5